src/reformulate.jl

```diff
@@ -1,82 +1,229 @@
+    ) where {T}
+    #TODO: need to pass _error to build_constraint
+    M = _calculate_tight_M(con)
+    if isinf(first(M))
+        M[1] = method.value
+    end
+    if isinf(last(M))
+        M[2] = method.value
     end
+    JuMP.add_constraint(model,
+        JuMP.build_constraint(error, 
+            con.func + first(M)*(1-bvar), 
+            _MOI.GreaterThan(con.set.value)
+        ),
+    )
+    JuMP.add_constraint(model,
+        JuMP.build_constraint(error, 
+            con.func - last(M)*(1-bvar), 
+            _MOI.LessThan(con.set.value)
+        ),
+    )
+end
+
+"""
+
+"""
+function _reformulate(
+    model::JuMP.Model, 
+    method::Hull,
+    con::JuMP.ScalarConstraint{JuMP.AffExpr, <: _MOI.LessThan}, 
+    lvar::LogicalVariableRef
+    )
+    #TODO: need to pass _error to build_constraint
+    con_func = _disaggregated_constraint(model, con, lvar)
+    bvar = model[Symbol(lvar,"_Bin")]
+    con_func.terms[bvar] = -con.set.upper
+
+    JuMP.add_constraint(model,
+        JuMP.build_constraint(error,
+            con_func,
+            _MOI.LessThan(0)
+        )
+    )
+end
+function _reformulate(
+    model::JuMP.Model, 
+    method::Hull,
+    con::JuMP.ScalarConstraint{JuMP.AffExpr, <: _MOI.GreaterThan}, 
+    lvar::LogicalVariableRef
+    )
+    #TODO: need to pass _error to build_constraint
+    con_func = _disaggregated_constraint(model, con, lvar)
+    bvar = model[Symbol(lvar,"_Bin")]
+    con_func.terms[bvar] = -con.set.lower
+    
+    JuMP.add_constraint(model,
+        JuMP.build_constraint(error,
+            con_func,
+            _MOI.GreaterThan(0)
+        )
+    )
+end
+function _reformulate(
+    model::JuMP.Model, 
+    method::Hull,
+    con::JuMP.ScalarConstraint{JuMP.AffExpr, <: _MOI.Interval}, 
+    lvar::LogicalVariableRef
+    )
+    #TODO: need to pass _error to build_constraint
+    con_func_GreaterThan = _disaggregated_constraint(model, con, lvar)
+    con_func_LessThan = copy(con_func_GreaterThan)
+    bvar = model[Symbol(lvar,"_Bin")]
+    con_func_GreaterThan.terms[bvar] = -con.set.lower
+    con_func_LessThan.terms[bvar] = -con.set.upper
+    
+    JuMP.add_constraint(model,
+        JuMP.build_constraint(error,
+            con_func_GreaterThan,
+            _MOI.GreaterThan(0)
+        )
+    )
+    JuMP.add_constraint(model,
+        JuMP.build_constraint(error,
+            con_func_LessThan,
+            _MOI.LessThan(0)
+        )
+    )
+end
+function _reformulate(
+    model::JuMP.Model, 
+    method::Hull,
+    con::JuMP.ScalarConstraint{JuMP.AffExpr, <: _MOI.EqualTo}, 
+    lvar::LogicalVariableRef
+    )
+    #TODO: need to pass _error to build_constraint
+    con_func = _disaggregated_constraint(model, con, lvar)
+    bvar = model[Symbol(lvar,"_Bin")]
+    con_func.terms[bvar] = -con.set.value
+
+    JuMP.add_constraint(model,
+        JuMP.build_constraint(error,
+            con_func,
+            _MOI.EqualTo(0)
+        )
+    )
 end
-reformulate_constraint(args...) = nothing
```

Comment:
It would be good to have a fallback for the `_reformulate` methods in case some other constraint type is intercepted.

Reply:
Added these two definitions to /src/reformulate.jl: 
# define fallbacks for other constraint types
_reformulate(model::JuMP.Model, method::Hull, args...) = nothing
_reformulate(model::JuMP.Model, method::BigM, args...) = nothing

Reply:
The problem with that is that such constraints will then be silently ignored. Better to throw an error if we are not going to reformulate them.

Reply:
done. let me know if there is a better way to do this. I did:
```
function _reformulate(
    model::JuMP.Model, 
    method::AbstractReformulationMethod, 
    con::JuMP.AbstractConstraint, 
    lvar::LogicalVariableRef
)
    error("$method reformulation for constraint $con is not supported yet.")
end
```

Reply:
That should do the trick